src/ext/readerResp.c

```diff
@@ -0,0 +1,256 @@
+}
+
+static RespRes readRespReplyLine(RespReaderCtx *ctx, RespReplyBuff *buffInfo) {
+
+    enum ProcessLineTypeStates {
+        PROC_LINE_START=0,
+        PROC_LINE_SEEK_CR=1,
+        PROC_LINE_FOUND_CR, // '\r'
+        PROC_LINE_FOUND_NEWLINE, // '\n'
+        PROC_LINE_END
+    };
+
+    while (buffInfo->at < buffInfo->len) {
+        switch (ctx->typeState) {
+
+            case PROC_LINE_START:
+                ctx->typeState = PROC_LINE_SEEK_CR;
+                /* fall-thru */
+            case PROC_LINE_SEEK_CR:
+                while (buffInfo->buff[(buffInfo->at)++] != '\r')
+                    if (buffInfo->at == buffInfo->len)
+                        return RESP_REPLY_PARTIAL;
+                ctx->typeState = PROC_LINE_FOUND_CR;
+                break;
+
+            case PROC_LINE_FOUND_CR:
+                if (buffInfo->buff[buffInfo->at] == '\n')
+                    ctx->typeState = PROC_LINE_FOUND_NEWLINE;
+                else
+                    ctx->typeState = PROC_LINE_SEEK_CR;
+                break;
+
+            case PROC_LINE_FOUND_NEWLINE:
+                ++(buffInfo->at);
+                /* fall-thru */
+            case PROC_LINE_END:
+                return RESP_REPLY_OK;
+
+        }
+    }
+    return RESP_REPLY_PARTIAL;
+}
+
+static RespRes readRespReplyError(RespReaderCtx *ctx, RespReplyBuff *buffInfo) {
+    int startAt = buffInfo->at;
+
+    /* can return either RESP_REPLY_PARTIAL or RESP_REPLY_OK */
+    RespRes res = readRespReplyLine(ctx, buffInfo);
+    if (res == RESP_REPLY_ERR)
+        return res;
+
+    int bytesProcessed =  buffInfo->at - startAt;
+    int errMsgFreeBytes = MAX_RESP_REPLY_ERR_MSG - ctx->errorMsgLen;
+    /* trim if error msg is bigger than errorMsg[] */
+    int lenToCopy = (bytesProcessed > errMsgFreeBytes) ? errMsgFreeBytes : bytesProcessed;
+
+    for (int i = 0 ; i < lenToCopy ; ++i)
+        ctx->errorMsg[ctx->errorMsgLen++] = buffInfo->buff[startAt + i];
+
+    /* If end of error message then end string with '\0' and  indicate for an error */
+    if (res == RESP_REPLY_OK) {
+        /* If errorMsg ends with '\r\n' then replace with '\0'.
+         * Else, it is too long and got trimmed. Replace last char with '\0' */
+        if (ctx->errorMsg[ctx->errorMsgLen-1] == '\n')
+            ctx->errorMsg[ctx->errorMsgLen - 2] = '\0';
+        else
+            ctx->errorMsg[ctx->errorMsgLen - 1] = '\0';
+
+        res = RESP_REPLY_ERR;
+    }
+
+    return res;
+}
+
+static RespRes readRespReplyBulk(RespReaderCtx *ctx, RespReplyBuff *buffInfo) {
+    UNUSED(buffInfo);
+    snprintf(ctx->errorMsg,sizeof(ctx->errorMsg),"Unexpected bulk reply");
+    return RESP_REPLY_ERR;
+}
+
+static RespRes readRespReplyAggregate(RespReaderCtx *ctx, RespReplyBuff *buffInfo) {
+    UNUSED(buffInfo);
+    snprintf(ctx->errorMsg,sizeof(ctx->errorMsg),"Unexpected aggregate reply");
+    return RESP_REPLY_ERR;
+}
```

Comment:
am i reviewing too early?

Reply:
No. As long as there are no commands, which sent by respToTcpLoader.c, that will cause to get back 
aggregated replies, I think it is better not support it. 

I do need TODO some effort to have better error reporting and reflect the failed command.

Reply:
at some point or another, we might end up sending a command that does return that.
but anyway, for now, let's add a comment saying why it's dead code..